janusgraph-dist/docker/docker-entrypoint.sh

```diff
@@ -32,7 +32,7 @@ if [ "$1" == 'janusgraph' ]; then
   mkdir -p ${JANUS_DATA_DIR} ${JANUS_CONFIG_DIR}
   cp conf/janusgraph-${JANUS_PROPS_TEMPLATE}-server.properties ${JANUS_PROPS}
   cp conf/janusgraph-server.yaml ${JANUSGRAPH_SERVER_YAML}
-  chown -R "$(id -u):$(id -g)" ${JANUS_DATA_DIR} ${JANUS_CONFIG_DIR}
+  chown "$(id -u):$(id -g)" ${JANUS_DATA_DIR} ${JANUS_CONFIG_DIR}
```

Comment:
I think I agree with the solution here.
However, I wanted to notice that the configuration is going to be overwritten here anyways. So, if the user changes some configs manually then they will be lost after the re-deployment. Not sure if that's the right behavior or not, but alternatively we could skip this whole section if  `${JANUS_DATA_DIR}` is present already (or maybe just use some file flag like `isInitialized`). In such case we won't run this section after the re-deployment at all. It just some thoughts, not sure if the solution I propose is viable or not.

Reply:
I'm not very proficient with shell scripting 😅, so wouldn't be comfortable adding extra logic tbh. Especially since this PR just aim at a quick fix. Can help to create something better in a separate PR of course, if you think it's worth doing.

Reply:
The current solution if fine by me. I'm not sure if users of this docker image need to have an ability to change configuration files manually or not. I don't use this image, so can't speak much about how it's better to be made.

Reply:
Users have 2 ways to configure the container:
1. use environment variables like `janusgraph.storage.hostname` which will be used to create the config (documented [here](https://docs.janusgraph.org/operations/container/#docker-environment-variables))
2. mount a directory to `/etc/opt/janusgraph`.

The mount option is documented [here](https://docs.janusgraph.org/operations/container/#mounted-configuration) and it mentions:
> You'll need to bind the files as read-only, however, if you do not wish to have the environment variables override the values in that file.

Does that clear things up for you, @porunov?

I have also just built the image locally with these changes and tried out both approaches just to make sure that it doesn't break anything.

Reply:
In that case the current solution in this PR is complete. Thank you for explanation.